home/views/api/leaderboard.py

```diff
@@ -0,0 +1,365 @@
+        # Json response template
+        json_response = {
+            "status": "success",
+            "message": "Leaderboard accessed successfully",
+            "payload": {
+                "account_id": device.device_id,
+                "leaderboard": [],
+            },
+        }
+
+        # def get(self, request, *args, **kwargs):
+        # user_stats_container = {}
+
+        contest = Contest.active()
+
+        # daily_walks, intentional_walks, _, _ = get_contest_walks(contest)
+
+        # # get the current/next Contest
+        leaderboard_list = []
+        # user_steps = {}
+        # for email, dw_row in daily_walks.items():
+        #     acct = Account.objects.values(*ACCOUNT_FIELDS).get(email=email)
+        #    # print("acct", acct)
+        #     accountid = acct.get("id")
+        #     dev_id = (
+        #         Account.objects.values("device")
+        #         .filter(id=accountid)
+        #         .reverse()[0]
+        #     )
+        #     #print("dev_id", dev_id)
+        #     dev = Device.objects.values(*DEVICE_FIELDS).filter(
+        #         account_id=accountid
+        #     )[0]
+        #    # print("device id", dev)
+        #    # print("iso d", dev["device_id"])
+        #     # print("id, ",Device.objects.get("device_id"))
+        #     if contest and acct["created"] > localize(contest.end):
+        #         continue
+
+        #     user_stats = user_stats_container.get(
+        #         email, dict(new_signup=False)
+        #     )
+        #     user_stats["account"] = acct
+        #     user_stats["dw_steps"] = dw_row["dw_steps"]
+        #     id = acct["id"]
+        #     # device = Device.objects.get(device_id = acct)
+        #     # print(device)
+
+        #     # id = device.device_id
+        #     # device = Device.objects.get(device_id=json_data["account_id"])
+        #     rank = -1
+        #     user_steps[id] = dw_row["dw_steps"]
+        #     leaderboard_list.append(
+        #         {"rank":rank, "id": dev["device_id"], "steps": user_steps[id]}
+        #     )
+        contest = Contest.active()
+        leaderboard_list = []
+        leaderboard_length = 10
+
+        # c = [(idx, item) for idx,item in enumerate(leaderboard_list, start=1)]
+
+        def build_dict(seq, key):
+            return list(
+                (dict(d, rank=rank + 1)) for (rank, d) in enumerate(seq)
+            )
+
+        leaderboard = Leaderboard.objects.all()
+        model_leaderboard_list = []
+        for participant in leaderboard:
+            leader_dict = {}
+            leader_dict["id"] = participant.device.device_id
+            leader_dict["steps"] = participant.steps
+
+            model_leaderboard_list.append(leader_dict)
+        # print("dict", model_leaderboard_list)
+
+        leaderboard_list = sorted(
+            model_leaderboard_list, key=lambda user: -user["steps"]
+        )
+
+        leaderboard_list = build_dict(leaderboard_list, key="steps")
```

Comment:
I don't think we need to iterate through the list to add rank numbers.

In sql, there's a `row_number()` function you can call to have the database generate them for every row in the result set. I think you'd use QuerySet `annotate` to add this to the results. Instead of querying for objects, query for `values` and specify the fields you want, so can add these annotations.  Look at the queries in my branch in admin.py...